orders/src/main/java/com/orders/crudyorders/model/Order.java

```diff
@@ -0,0 +1,83 @@
+package com.orders.crudyorders.model;
+
+import javax.persistence.*;
+
+public class Order
+{
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(nullable = false)
+    private long ordnum;
+
+    private double ordamount, advanceamount;
+
+    @OneToMany
+    @JoinColumn(name = "customer", nullable = false)
+    private long custcode;
+    // this is a long foreign key
+
+    @OneToMany
+    @JoinColumn(name = "agent", nullable = false)
+    private long agentcode;
```

Comment:
• Set this as a many-to-many relationship and the name for the `@JoinColumn` annotation must be the same as the ID field of the type you are referencing (`agentcode` here)
• Foreign keys aren't done with ID data types like `long` here; just directly use the class of the object you are referencing (in this case, `Agent`)
• Using the `@JsonIgnoreProperties` annotation, prevent the `customers` and `orders` properties of your `Agent` object from being displayed (or else the agent's customers will display the agent again, and so forth in an infinite loop) - this will be sent as an array of strings with the notation `({"property1", "property2"})`
• Don't forget to include this property in your constructor, getters and setters

The directions above should result in:
```
@ManyToOne
@JoinColumn(name = "agentcode", nullable = false)
@JsonIgnoreProperties({"customers", "orders"})
private Agent agent;
```